followthemoney/types/identifier.py

```diff
@@ -22,6 +27,19 @@ class IdentifierType(PropertyType):
     matchable = True
     pivot = True
 
+    def clean_text(
+        self,
+        text: str,
+        fuzzy: bool = False,
+        format: Optional[str] = None,
+        proxy: Optional["EntityProxy"] = None,
+    ) -> Optional[str]:
+        if format is not None and format in FORMATS:
+            type_ = FORMATS[format]
+            return type_.normalize(text)
+        return text
```

Comment:
Two questions out of curiosity:

* Validating concrete identifier formats is a new feature, right? Actually, if I don’t miss anything, the only use for property type formats so far was the date type, right?

* I vaguely remember that we talked about adding the identifier format to the schema, so e.g. `Thing:wikidataId` would be an `identifier` in the `wikidata` format? Is that still something you’d like to add? I think it’s a good idea, but in that case would like to check on our side if/where these stricter validations could cause problems (e.g. in scrapers).

Reply:
Nice catch! So this is new functionality, but for the moment it would only come into effect in a mapping (that's the only place the `format` is set from) which would specify `format` for an identifier field. So it's completely opt-in. 

The other option - making properties themselves know format - makes a ton of sense semantically (they're literally called `wikidataId`, `swiftBic`, `imoCode`, `innCode`, etc.) but it would create a breaking change in the sense that old mappings might produce much less data because all of a sudden a whole ton of INNs are recognised as invalid. There's maybe an argument for that to be a good thing, but it'd definitely be a surprising thing.

Maybe there's even a way of half-doing it: some props, like `wikidataId` have strong expectations, and we could introduce it there. By doing that we could also make IBAN fields from `type: iban` into `type: identifier, format: iban` which makes a bit more sense. 

WDYT?

Reply:
> but it would create a breaking change in the sense that old mappings might produce much less data because all of a sudden a whole ton of INNs are recognised as invalid. There's maybe an argument for that to be a good thing, but it'd definitely be a surprising thing.

Personally, I’d say that’s a good thing as long as there’s maybe some logging in the mapping logic when values are removed because they are invalid, allowing the monitoring of e.g. Memorious scrapers for validation issues.

But I’ve also forwarded this thread to our data team because they’d be much more qualified to assess the potential impact on our scraper fleet.

Reply:
The other use case where validation is relevant from Aleph’s perspective is OCR/pattern extraction: When extracting IBANs from scanned documents, there might be OCR issues that e.g. cause the extracted IBAN to not match validation. While such an invalid IBAN might not be as useful for cross referencing, there might still be value in storing and exposing the extracted IBAN to users.

This isn’t really relevant to this PR, as the PR doesn’t change how IBANs are validated (there’s just an additional layer of abstraction), but something to keep in mind for the future and possibly when adding other identifier formats. (For most of the other identifier formats implemented in rigour, it’s probably much more likely that they are extracted from company registries etc. rather than from a scanned document, but I might be incorrect here.)